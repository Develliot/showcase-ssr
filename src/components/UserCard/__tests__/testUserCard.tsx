import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { UserCard } from '../';

const mockUser = {
    email: 'angela.castro@example.com',
    location: {
        street: {
            number: '8267',
            name: 'Grove Road',
        },
        city: 'Stevenage',
        state: 'Durham',
        postcode: 'BD18 8JD',
        coordinates: {
            latitude: 61.2773,
            longitude: 45.9598,
        },
    },
    login: {
        uuid: '1255d3c5-a9c8-414e-8eb5-127478a19f62',
    },
    name: { first: 'Angela', last: 'Castro' },
    picture: {
        large: 'https://randomuser.me/api/portraits/women/45.jpg',
    },
};

it('renders correctly', () => {
    const wrapper: ShallowWrapper = shallow(
        <UserCard user={mockUser} color='#A93F55' />
    );
    expect(wrapper).toMatchSnapshot();
});
