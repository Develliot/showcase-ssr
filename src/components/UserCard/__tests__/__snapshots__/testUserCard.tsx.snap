// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styles__UserCardWrapper>
  <Layout__RowCenter>
    <styles__CoverColor
      color="#A93F55"
    >
      <Layout__RowCenter>
        <styles__ImageWrapper>
          <styles__ImageContainer
            alt="User image"
            src="https://randomuser.me/api/portraits/women/45.jpg"
          />
        </styles__ImageWrapper>
      </Layout__RowCenter>
    </styles__CoverColor>
  </Layout__RowCenter>
  <styles__BottomHalf>
    <Layout__VerticalSpacer
      size="small"
    />
    <Typography__H3>
      Angela
       
      Castro
    </Typography__H3>
    <Layout__VerticalSpacer
      size="small"
    />
    <Typography__Paragraph
      ellipsis={true}
    >
      <Typography__Span
        bold={true}
      >
        Email: 
      </Typography__Span>
       
      angela.castro@example.com
    </Typography__Paragraph>
    <Typography__Paragraph
      ellipsis={true}
    >
      <Typography__Span
        bold={true}
      >
        Street: 
      </Typography__Span>
       
      <Typography__Span>
        8267
         
        Grove Road
      </Typography__Span>
    </Typography__Paragraph>
    <Typography__Paragraph
      ellipsis={true}
    >
      <Typography__Span
        bold={true}
      >
        City: 
      </Typography__Span>
       
      Stevenage
    </Typography__Paragraph>
    <Typography__Paragraph
      ellipsis={true}
    >
      <Typography__Span
        bold={true}
      >
        State: 
      </Typography__Span>
       
      Durham
    </Typography__Paragraph>
    <Typography__Paragraph
      ellipsis={true}
    >
      <Typography__Span
        bold={true}
      >
        Postcode: 
      </Typography__Span>
       
      BD18 8JD
    </Typography__Paragraph>
  </styles__BottomHalf>
</styles__UserCardWrapper>
`;
